sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2015,7 +2015,7 @@ pi_result piDevicePartition(pi_device Device,
         OutDevices[I] = PiSubDevice.get();
         Platform->PiDevicesCache.push_back(std::move(PiSubDevice));
         // save pointers to sub-devices for quick retrieval in the future.
-        Device->SubDevices.push_back(Dev);
+        Device->SubDevices.push_back(OutDevices[I]);
```

Comment:
The OutDevices is "owned" by the caller, so it's unnatural to have that value saved. Could you change to save PiSubDevice instead?

Reply:
The values being pushed here will become outputs on the next call. And, the _values_ in OutDevice[I] are being saved, not OutDevice itself, but fine, I'll change it.

Reply:
Note, that because PiSubDevice is "moved", slight rearrangement of lines was also needed.